topsort-api-v1.yml

```diff
@@ -183,32 +194,6 @@ components:
             $ref: '#/components/schemas/TopsortError'
 
   schemas:
-    AuctionRequest:
```

Comment:
Could you check how does the generated code looks like. In general we need to avoid having anonymous types as those typically end up with random names, hence being unusable.

Reply:
It generates an empty type `AuctionRequest` which is used as the schema for `CreateAuction` and generates an `AnyOfAuctionRequest` model which is never used. In the previous iteration (without this model), the `AnyOfAuctionRequest` is named `AnyOfauctionsBody`.

Okay, I ended up leaving this as the first option, but changed the OpenAPI labels not to render the 3 SchemaDefinition models in Redocly that were initially removed since the dropdowns for models with discriminators are broken.

Reply:
I don't think client code can be generated automatically with the API as it is now; the union types make this too complex (and impossible for languages such as Go unless we allow any fields). We need to avoid anyOf/oneOf in future versions if we care about automating code generation.

Reply:
At the same time, for this version we should not be lax on the definitions otherwise people will send stuff that's not allowed by the engine but allowed according to our API docs.